src/driver-sinowealth.c

```diff
@@ -1465,6 +1465,18 @@ static const struct sinowealth_device_data sinowealth_supported_devices[] = {
 		.button_count = 6,
 	},
 
+	/* 2581:1007 devices. */
+	{
+		.fw_version = "9677",
+		.device_name = "Marvo Scorpion G961",
```

Comment:
regarding the commit message: do we have a name match yet? If not, that could simply be added to avoid that issue.

Reply:
What do you mean a `name match`? This `device_name` field is not actually reported by the mouse, it's something I added to distinguish between all of these device entries. Used device is actually detected by `fw_version` field.

Reply:
oh, that name isn't even in the hidraw? What does the device report itself as?

Reply:
All of them report themselves as `SINOWEALTH Wired Gaming Mouse`.

Reply:
well, that's reasonably useless... in that case we'd need some extra matching logic so we can identify the driver based on some matches, but punt the exact match to the driver. Which is going to need some... intermediate plans, so for now it's easier to go with what you have.